src/emit/mod.rs

```diff
@@ -5534,6 +5534,198 @@ pub trait TargetRuntime<'a> {
+                    .build_pointer_cast(
+                        l,
+                        bin.context.i32_type().ptr_type(AddressSpace::Generic),
+                        "left",
+                    )
+                    .into(),
+                bin.builder
+                    .build_pointer_cast(
+                        r,
+                        bin.context.i32_type().ptr_type(AddressSpace::Generic),
+                        "right",
+                    )
+                    .into(),
+                bin.builder
+                    .build_pointer_cast(
+                        o,
+                        bin.context.i32_type().ptr_type(AddressSpace::Generic),
+                        "output",
+                    )
+                    .into(),
+                bin.context
+                    .i32_type()
+                    .const_int(mul_bits as u64 / 32, false)
+                    .into(),
+            ],
+            "",
+        );
+
+        let res = bin.builder.build_load(o, "mul");
+
+        let error_block = bin.context.append_basic_block(function, "error");
+        let return_block = bin.context.append_basic_block(function, "return_block");
+
+        // Extract sign bit of the operands and the result
+        let left_sign_bit = self.extract_sign_bit(bin, left, left.get_type());
+        let right_sign_bit = self.extract_sign_bit(bin, right, right.get_type());
+        let res_sign_bit = if mul_bits == bits {
+            // If no extension took place, get the leftmost bit(sign bit).
+            self.extract_sign_bit(bin, res.into_int_value(), res.into_int_value().get_type())
+        } else {
+            // If extension took place, truncate the result to the type of the operands then extract the leftmost bit(sign bit).
+            self.extract_sign_bit(
+                bin,
+                bin.builder
+                    .build_int_truncate(res.into_int_value(), left.get_type(), ""),
+                left.get_type(),
+            )
+        };
+
+        // If the operands sign is the same
+        let same_signs =
+            bin.builder
+                .build_int_compare(IntPredicate::EQ, left_sign_bit, right_sign_bit, "");
+
+        // If operands sign is the same and the result is positive
+        let same_operands_and_positive_res =
+            bin.builder
+                .build_and(same_signs, bin.builder.build_not(res_sign_bit, ""), "");
+
+        // If operands are not the same and the result is negative
+        let diff_operands_and_neg_res =
+            bin.builder
+                .build_and(bin.builder.build_not(same_signs, ""), res_sign_bit, "");
+        let left_is_zero =
+            bin.builder
+                .build_int_compare(IntPredicate::EQ, left, left.get_type().const_zero(), "");
+        let right_is_zero = bin.builder.build_int_compare(
+            IntPredicate::EQ,
+            right,
+            right.get_type().const_zero(),
+            "",
+        );
```

Comment:
There are two versions of mul32, void mul32 and mul32 with ovf.
In signed ovf detection, I call void mul32 and check for changes in the sign, that's it. I may go with calling mul32 with ovf but it is not needed here.
So, answering your question, it is indeed easier if I check the result of __mul32, but the result may have overflowed and res is all zeroes. 
As of why I call void mul32 and not the ovf version, that is because void mul32 is less expensive.